### PR TITLE
test: 799 spatial e2e test parity

### PR DESCRIPTION
**Pull request by tihuan**
1. Add `super-cool-spatial.cxg` as part of the regular e2e test suite via a loop
2. Add `super-cool-spatial.cxg` specific data in `data.ts`
3. REMOVE `truncation-test.cxg` from `data.ts`, since it doesn't seem used
4. Create both `darwin` and `linux` snapshot files via this command (THANKS ChatGPT 🔥 ):

```ts
for file in __tests__/e2e/e2e.test.ts-snapshots/*-darwin.txt; do
    cp "$file" "${file/-darwin.txt/-linux.txt}"
done
```

### CODE REVIEW

File: client/__tests__/e2e/e2e.test.ts [hunk 1]
```diff
@@ -77,773 +81,842 @@ const setToRemoveFrom = "empty_this_geneset";
...
+    });
 
-      const newGraph = await page.getByTestId("graph-wrapper");
-      const newDistance =
-        (await newGraph.getAttribute("data-camera-distance")) ?? "-1";
-      expect(parseFloat(newDistance)).toBe(scaleMax);
-    },
-    { page }
-  );
-});
```
Comment (tihuan) [inline client/__tests__/e2e/e2e.test.ts:475]: no change - just got bumped down

File: client/__tests__/e2e/e2e.test.ts [hunk 2]
```diff
@@ -77,773 +81,842 @@ const setToRemoveFrom = "empty_this_geneset";
...
+      );
 
-  await drag(
-    "layout-graph",
-    lassoSelection.start,
-    lassoSelection.end,
-    page,
-    true
-  );
-  await expect(page.getByTestId("lasso-element")).toBeVisible();
```
Comment (tihuan) [inline client/__tests__/e2e/e2e.test.ts:494]: no change

File: client/__tests__/e2e/e2e.test.ts [hunk 3]
```diff
@@ -77,773 +81,842 @@ const setToRemoveFrom = "empty_this_geneset";
...
 
-  const initialCount = await getCellSetCount(1, page);
+          const newGraph = await page.getByTestId("graph-wrapper");
+          const newDistance =
+            (await newGraph.getAttribute("data-camera-distance")) ?? "-1";
+          expect(parseFloat(newDistance)).toBe(scaleMax);
+        },
+        { page }
+      );
+    });
```
Comment (tihuan) [inline client/__tests__/e2e/e2e.test.ts:465]: wrapped in `tryUntil()`

File: client/__tests__/e2e/e2e.test.ts [hunk 4]
```diff
@@ -77,773 +81,842 @@ const setToRemoveFrom = "empty_this_geneset";
...
-  expect(modeSwitchCount).toBe(initialCount);
-});
+          await drag(
+            "layout-graph",
+            lassoSelection.start,
+            lassoSelection.end,
+            page,
+            true
+          );
+          await expect(page.getByTestId("lasso-element")).toBeVisible();
```
Comment (tihuan) [inline client/__tests__/e2e/e2e.test.ts:487]: wrapped in `tryUntil()`

File: client/__tests__/e2e/e2e.test.ts [hunk 5]
```diff
@@ -77,773 +81,842 @@ const setToRemoveFrom = "empty_this_geneset";
...
-    });
-    test("color by mean expression", async ({ page }, testInfo) => {
-      await setup(option, page);
-      await createGeneset(meanExpressionBrushGenesetName, page);
-      await addGeneToSetAndExpand(meanExpressionBrushGenesetName, "SIK1", page);
-
-      await colorByGeneset(meanExpressionBrushGenesetName, page);
-      await assertColorLegendLabel(meanExpressionBrushGenesetName, page);
-      await snapshotTestGraph(page, testInfo);
     });
```
Comment (tihuan) [inline client/__tests__/e2e/e2e.test.ts]: no change - just nested

File: client/__tests__/e2e/e2e.test.ts [hunk 6]
```diff
@@ -77,773 +81,842 @@ const setToRemoveFrom = "empty_this_geneset";
...
+          // this test will take longer if we're running against a deployment
+          if (runningAgainstDeployment) test.slow();
 
-      // (thuang): We need to assert Pop2 geneset is expanded, because sometimes
-      // the click is so fast that it's not registered
-      await tryUntil(
-        async () => {
-          await page.getByTestId("pop-1-geneset-expand").click();
-          await page.getByTestId("pop-2-geneset-expand").click();
+          await setup({ option, page, url });
```
Comment (tihuan) [inline client/__tests__/e2e/e2e.test.ts:616]: no change

File: client/__tests__/e2e/e2e.test.ts [hunk 7]
```diff
@@ -77,773 +81,834 @@ const setToRemoveFrom = "empty_this_geneset";
...
+          await page
+            .getByTestId(`${category}:category-select`)
+            .click({ force: true });
 
-      genesHTML = await page.getByTestId("gene-set-genes").innerHTML();
+          const cellset1Selector = page.getByTestId(
+            `categorical-value-select-${category}-${cellset1.cellType}`
+          );
+          const cellset2Selector = page.getByTestId(
+            `categorical-value-select-${category}-${cellset2.cellType}`
```
Comment (tihuan) [inline client/__tests__/e2e/e2e.test.ts:630]: create selectors to keep the code DRY

File: client/__tests__/e2e/e2e.test.ts [hunk 8]
```diff
@@ -77,773 +81,834 @@ const setToRemoveFrom = "empty_this_geneset";
...
+              await setup({ option, page, url });
+              await createGeneset(genesetToDeleteName, page);
+              await deleteGeneset(genesetToDeleteName, page);
+              await assertUndoRedo(
+                page,
+                async () => assertGenesetExists(genesetToDeleteName, page),
+                async () => assertGenesetDoesNotExist(genesetToDeleteName, page)
+              );
+            },
+            { page }
```
Comment (tihuan) [inline client/__tests__/e2e/e2e.test.ts:739]: no change

File: client/__tests__/e2e/e2e.test.ts [hunk 9]
```diff
@@ -77,773 +81,834 @@ const setToRemoveFrom = "empty_this_geneset";
...
+          );
+        });
+      });
 
-      await setup(option, page);
-      await createGeneset(
-        genesetToCheckForDescription,
-        page,
-        genesetDescriptionString
-      );
```
Comment (tihuan) [inline client/__tests__/e2e/e2e.test.ts:737]: no change

File: client/__tests__/e2e/e2e.test.ts [hunk 10]
```diff
@@ -77,773 +81,834 @@ const setToRemoveFrom = "empty_this_geneset";
...
+          const histBrushableAreaId = `histogram-${geneToBrushAndColorBy}-plot-brushable-area`;
+
+          const coords = await calcDragCoordinates(
+            histBrushableAreaId,
+            {
+              x1: 0.25,
+              y1: 0.5,
+              x2: 0.55,
+              y2: 0.5,
+            },
```
Comment (tihuan) [inline client/__tests__/e2e/e2e.test.ts:798]: no change

File: client/__tests__/e2e/e2e.test.ts [hunk 11]
```diff
@@ -77,773 +81,834 @@ const setToRemoveFrom = "empty_this_geneset";
...
+        test("open gene info card and hide/remove", async ({ page }) => {
+          await setup({ option, page, url });
+          await addGeneToSearch(geneToRequestInfo, page);
+
+          await tryUntil(
+            async () => {
+              await requestGeneInfo(geneToRequestInfo, page);
+              await assertGeneInfoCardExists(geneToRequestInfo, page);
+            },
+            { page }
```
Comment (tihuan) [inline client/__tests__/e2e/e2e.test.ts:852]: no change

File: client/__tests__/e2e/e2e.test.ts [hunk 12]
```diff
@@ -77,773 +81,834 @@ const setToRemoveFrom = "empty_this_geneset";
...
-      await colorByGene("SIK1", page);
-      await assertColorLegendLabel("SIK1", page);
-    });
-    test("delete gene from geneset and undo/redo", async ({ page }) => {
-      /**
-       * (thuang): Test is flaky, so we need to retry until it passes
-       */
-      await tryUntil(
-        async () => {
-          if (option.withSubset) return;
```
Comment (tihuan) [inline client/__tests__/e2e/e2e.test.ts:809]: no change

File: client/__tests__/e2e/e2e.test.ts [hunk 13]
```diff
@@ -77,773 +81,834 @@ const setToRemoveFrom = "empty_this_geneset";
...
-      }, { page });
-
-      await tryUntil(async () => {
-        await removeGeneInfo(page);
-        await assertGeneInfoDoesNotExist(geneToRequestInfo, page);
-      }, { page });
-    });
+        });
+      });
+    }
```
Comment (tihuan) [inline client/__tests__/e2e/e2e.test.ts:872]: no change

File: client/__tests__/e2e/e2e.test.ts [hunk 14]
```diff
@@ -77,773 +81,834 @@ const setToRemoveFrom = "empty_this_geneset";
...
+      const categoryRows = await page
+        .getByTestId(`category-truncate`)
+        .getByTestId("categorical-row")
+        .all();
+
+      expect(Object.keys(categoryRows).length).toBe(1001);
+    },
+    { page }
+  );
+});
```
Comment (tihuan) [inline client/__tests__/e2e/e2e.test.ts:892]: this is the truncate test that got moved outside of the dataset loop

File: client/__tests__/e2e/e2e.test.ts [hunk 15]
```diff
@@ -77,773 +81,834 @@ const setToRemoveFrom = "empty_this_geneset";
...
+
+/* eslint-enable no-await-in-loop -- await in loop is needed to emulate sequential user actions  */
+
+async function setup({
+  option: { withSubset },
+  page,
+  url,
```
Comment (tihuan) [inline client/__tests__/e2e/e2e.test.ts:899]: this function now takes dataset `url`

**Comment by tihuan** [comment]
@seve ready for review now 🙏 Thank you!

File: client/__tests__/e2e/data.ts [hunk 16]
```diff
@@ -157,16 +143,188 @@ export const datasets = {
...
+    brushOnGenesetMean: {
+      default: "131",
+      withSubset: "111",
+    },
+    expandGeneAndBrush: {
+      default: "109",
+      withSubset: "94",
+    },
   },
-  "truncation-test.cxg": {
```
Comment (seve) [inline client/__tests__/e2e/data.ts:161]: did we mean to remove the truncation test?

**Review by seve — APPROVED** [review]

**Pull request merged**